File: app/src/main/java/com/tryden/leetcode/java/easy/Sqrt.java
package com.tryden.leetcode.java.easy;

/*
 *
 * [69] Sqrt(x)
 *
 * https://leetcode.com/problems/sqrtx/description/
 *
 */

public class Sqrt {
    public static void main(String[] args) {
        int n = 98;
    }

    public static int mySqrt(int x) {
        long r = (x+1) / 2;
        while (r*r > x){
            r = (r + x/r) / 2;
            if (r < 0){
                r = r*-1;
            }
        }
        return (int) r;
    }
}

File: app/src/main/java/com/tryden/leetcode/kotlin/easy/Sqrt.kt
package com.tryden12.leetcode_kotlin.easy

/*
 *
 * [69] Sqrt(x)
 *
 * https://leetcode.com/problems/sqrtx/description/
 *
 */


    fun main() {
        println(mySqrt(8))
    }

    fun mySqrt(x: Int): Int {
        if(x < 2) return x

        var left = 2
        var right = x / 2

        while(left <= right) {
            var midd = left + (right - left) / 2
            var squaredMidd: Long = midd.toLong() * midd.toLong()


            if(squaredMidd > x) right = midd - 1
            else if (squaredMidd < x) left = midd + 1
            else return midd
        }

        return right
    }


